lib/Graph/Grad.cpp

```diff
@@ -275,6 +273,40 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
       continue;
     }
 
+    if (N->getKind() == Kind::GatherNodeKind) {
+      GatherNode *GN = cast<GatherNode>(N);
+      // Get gradient.
+      NodeValue Result = GN->getResult();
+      NodeValue OutputG = map.getGradient(Result);
+      // Get Data & Indices.
+      NodeValue Data = GN->getData();
+      NodeValue Indices = GN->getIndices();
+
+      // Reshape indices into a one-dimensional Tensor (Vector).
+      std::vector<size_t> IndicesDims{Indices.getType()->size()};
+      auto *RI = G->createReshape("reshape.indices.grad", Indices, IndicesDims);
```

Comment:
Ah, so the organization of the indices doesn't matter to the gradient computation. Nice trick!